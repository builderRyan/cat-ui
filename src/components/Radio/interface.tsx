import * as React from 'react'

export interface RadioInnerProps {
  checked?: boolean;
  disabled?: boolean;
}

export interface RadioInputProps {
  disabled?: boolean;
}

export interface RadioSlotProps {
  disabled?: boolean;
}
