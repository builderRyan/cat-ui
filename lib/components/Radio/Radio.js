"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_1 = require("react");
var classnames_1 = require("classnames");
var styled_1 = require("./styled");
var Radio = (function (_super) {
    __extends(Radio, _super);
    function Radio() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.handleChange = function (e) {
            console.log('handleChange');
        };
        _this.handleClick = function (e) {
            var _a = _this.props, disabled = _a.disabled, onChange = _a.onChange;
            if (disabled) {
                return;
            }
            onChange && onChange({
                target: __assign({}, _this.props, { checked: e.target.checked }),
                stopPropagation: function () {
                    e.stopPropagation();
                },
                preventDefault: function () {
                    e.preventDefault();
                },
                nativeEvent: e.nativeEvent
            });
        };
        return _this;
    }
    Radio.prototype.render = function () {
        var _a = this.props, className = _a.className, style = _a.style, disabled = _a.disabled, readOnly = _a.readOnly, checked = _a.checked, value = _a.value, children = _a.children;
        var classes = classnames_1.default('hmly-radio', className);
        return (React.createElement(styled_1.StyledRadioBox, { className: classes, style: style },
            React.createElement(styled_1.StyledRadio, null,
                React.createElement(styled_1.RadioInner, { checked: checked, disabled: disabled }),
                React.createElement(styled_1.RadioInput, { value: value, checked: checked, disabled: disabled, readOnly: readOnly, onClick: this.handleClick, onChange: this.handleChange })),
            React.createElement(styled_1.RadioSlot, { disabled: disabled }, children)));
    };
    return Radio;
}(react_1.Component));
exports.default = Radio;
//# sourceMappingURL=Radio.js.map