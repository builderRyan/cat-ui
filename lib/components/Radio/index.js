"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Radio_1 = require("./Radio");
var RadioGroup_1 = require("./RadioGroup");
Radio_1.default.Group = RadioGroup_1.default;
exports.default = Radio_1.default;
//# sourceMappingURL=index.js.map