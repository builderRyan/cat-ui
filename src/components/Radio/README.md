# Radio 单选框

## Example

```tsx
<Radio.Group value={value} onChange={(value) => setValue(value)}>
  <Radio value='A'>A</Radio>
  <Radio value='B'>B</Radio>
  <Radio value='C'>C</Radio>
  <Radio value='D'>D</Radio>
</Radio.Group>
```

## Radio API

|      属性      |                        类型                        | 默认值  |      描述      | 是否必填 |
| :------------: | :------------------------------------------------: | :-----: | :------------: | :------: |
|     preifx     |                      `string`                      |  `cat`  |    类名前缀    |    -     |
|   className    |                      `string`                      |    -    |      类名      |    -     |
|     style      |               `React.CSSProperties`                |    -    |    行内样式    |    -     |
|    disabled    |                     `boolean`                      | `false` |    是否禁用    |    -     |
| defaultChecked |                     `boolean`                      |    -    | 默认是否被选中 |    -     |
|    checked     |                     `boolean`                      |    -    |    是否选中    |    -     |
|     value      |                       `any`                        |    -    |       值       |    是    |
|    onChange    | `(e: React.ChangeEvent<HTMLInputElement>) => void` |    -    |  改变时的回调  |    -     |

## Radio.Group API

|      属性      |          类型          | 默认值  |      描述      | 是否必填 |
| :------------: | :--------------------: | :-----: | :------------: | :------: |
|   className    |        `string`        |    -    |      类名      |    -     |
|     style      | `React.CSSProperties`  |    -    |    行内样式    |    -     |
|    disabled    |       `boolean`        | `false` |    是否禁用    |    -     |
| defaultChecked |       `boolean`        |    -    | 默认是否被选中 |    -     |
|    checked     |       `boolean`        |    -    |    是否选中    |    -     |
|     value      |         `any`          |    -    | 当前选中项的值 |    是    |
|    onChange    | `(value: any) => void` |    -    |  改变时的回调  |    -     |
