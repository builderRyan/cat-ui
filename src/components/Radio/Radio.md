Example

```jsx
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      activeId1: 3
    }
    this.onRadioChange = this.onRadioChange.bind(this)
  }

  onRadioChange (id) {
    this.setState({ activeId1: id })
  }

  render () {
    const { activeId1 } = this.state
    return (<div className='radio-box'>
      <h3>单独使用</h3>
      <Radio value={1}>1</Radio>
      <Radio checked value={2}>2</Radio>
      <h3>Radio.Group</h3>
      <Radio.Group value={activeId1} onChange={this.onRadioChange}>
        <Radio value={1}>1</Radio>
        <Radio value={2}>2</Radio>
        <Radio value={3}>3</Radio>
        <Radio disabled value={4}>4</Radio>
        <Radio value={5} shape='hook'>5</Radio>
      </Radio.Group>

      <style>
        {`
          .radio-box {
            padding: 5px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```
