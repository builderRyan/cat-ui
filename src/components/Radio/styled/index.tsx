import styled from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'
import { RadioInnerProps } from '../interface'

const StyledRadioBox = styled.div`
  position: relative;
  padding: 0;
  list-style: none;
  display: inline-block;
  line-height: 1.5;
  white-space: nowrap;
  cursor: pointer;
  box-sizing: border-box;
  margin-top: -2px;
  color: #767676;
  font-weight: 600;
  & + & {
    margin-left: 40px;
  }
`;

const StyledRadio = styled.span`
  position: relative;
  display: inline-block;
  margin: 0;
  padding: 0;
  list-style: none;
  outline: none;
  white-space: nowrap;
  cursor: pointer;
  line-height: 1;
  vertical-align: text-top;
`

const RadioInner = styled.span`
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 20px;
  height: 20px;
  border: 1px solid #313646;
  border-radius: 50%;
  transition: all .3s;
  box-sizing: border-box;
  background-color: #f5f5f5;
  &::after {
    position: absolute;
    width: 10px;
    height: 10px;
    left: 4px;
    top: 4px;
    border-radius: 50%;
    display: table;
    border-top: 0;
    border-left: 0;
    content: " ";
    background-color: #313646;
    opacity: ${(props: RadioInnerProps) => props.checked ? 1: 0};
    transition: all .3s cubic-bezier(.78,.14,.15,.86);
    box-sizing: border-box;
  }
`

const RadioInput = styled.input`
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
  padding: 0;
  box-sizing: border-box;
`;

const RadioSlot = styled.span`
  display: inline-block;
  word-wrap: break-word;
  white-space: pre-wrap;
  margin-top: -2px;
  padding-left: 13px;
  font-size: 14px;
  line-height: 24px;
  cursor: pointer;
  box-sizing: border-box;
  vertical-align: text-top;
`
const StyledRadioGroup = styled.div``

export { StyledRadioBox, StyledRadio, RadioInput, RadioInner, RadioSlot, StyledRadioGroup }
