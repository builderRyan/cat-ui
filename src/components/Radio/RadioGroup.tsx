import * as React from 'react'
import { Component, cloneElement } from 'react'
import classNames from 'classnames'
import { StyledRadioGroup } from './styled'

export interface RadioGroupProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 是否禁用 */
  disabled?: boolean;
  /** 是否只读 */
  readOnly?: boolean;
  /** 是否选中 */
  checked?: boolean;
  /** 值 */
  value?: any;
  /** 选项变化时的回调函数	 */
  onChange?: any;
}

/**
 * RadioGroup
 */
class RadioGroup extends Component<RadioGroupProps, any> {

  handleChange = (e: any) => {
    console.log('handleChange')
  }

  handleClick = (e: any) => {
    console.log('handleClick')
  }

  onRadioChange = (e: any) => {
    const { value } = e.target
    const onChange = this.props.onChange
    console.log('RadioGroup:onRadioChange', value)
    onChange && onChange(value)
  }

  render() {
    const self = this
    const { className, style, disabled, readOnly, checked, value, children } = this.props
    const classes = classNames('hmly-radio-group', className)
    const radios = React.Children.map(children, (element: any, index) => {
      if (!element) { return element }
      return cloneElement(element, {
        onChange: self.onRadioChange,
        key: index,
        checked: value === element.props.value
      })
    })

    return (
      <StyledRadioGroup
        className={classes}
        style={style}>
        {radios}
      </StyledRadioGroup>
    )
  }
}

export default RadioGroup
