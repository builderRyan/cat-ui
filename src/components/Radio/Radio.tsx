import React from 'react';
import { Component } from 'react';
import classNames from 'classnames';
import RadioGroup from './RadioGroup';
import PropTypes from 'prop-types';

export interface RadioProps {
  id?: string;
  name?: string;
  /**
   * 类名前缀
   */
  prefix: string;
  /** 
   * 类名
   */
  className?: string;
  /** 
   * 样式
   */
  style?: React.CSSProperties;
  /** 
   * 是否禁用
   */
  disabled: boolean;
  /**
   * 默认是否选中
   */
  defaultChecked: boolean;
  /** 
   * 是否选中
   */
  checked?: boolean;
  /** 
   * 值
   */
  value: any;
  /**
   * 点击时的回调
   */
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface RadioState {
  checked: boolean;
}

/**
 * 单选框
 */
class Radio extends Component<RadioProps, RadioState> {
  static propTypes = {
    prefix: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    disabled: PropTypes.bool,
    defaultChecked: PropTypes.bool,
    checked: PropTypes.bool,
    value: PropTypes.any,
  }

  static defaultProps = {
    prefix: 'cat',
    disabled: false,
    defaultChecked: false,
  };

  static Group: typeof RadioGroup;

  static getDerivedStateFromProps(nextProps: RadioProps) {
    if ('checked' in nextProps) {
      return { checked: nextProps.checked };
    }
    return null;
  }

  constructor(props: RadioProps) {
    super(props);
    let checked = false;
    if ('checked' in props) {
      checked = !!props.checked;
    } else {
      checked = !!props.defaultChecked;
    }
    this.state = {
      checked,
    };
  }

  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { disabled, onChange } = this.props;
    if (disabled) {
      return;
    }

    this.setState({ checked: !this.state.checked });
    onChange && onChange(e);
  };

  render() {
    const {
      id,
      name,
      className,
      style,
      disabled,
      children,
      prefix,
      value,
    } = this.props;
    const { checked } = this.state;

    const classPrefix = `${prefix}-radio`;
    const classes = classNames(
      `${classPrefix}-wrapper`,
      {
        [`${classPrefix}-wrapper--disabled`]: disabled,
      },
      className,
    );

    return (
      <label className={classes} style={style}>
        <span
          className={classNames(
            `${classPrefix}__inner-wrapper`,
          )}
        >
          <span
            className={classNames(`${classPrefix}__inner`, {
              [`${classPrefix}__inner--disabled`]: disabled,
              [`${classPrefix}__inner--checked`]: checked
            })}
          />
          <input
            className={classNames(classPrefix, {
              [`${classPrefix}--disabled`]: disabled
            })}
            type='radio'
            name={name}
            id={id}
            value={value}
            checked={checked}
            disabled={disabled}
            onChange={this.handleChange}
          />
        </span>
        {children && (
          <span
            className={classNames(
              'cat-radio__label',
              {
                [`cat-radio__label--disabled`]: disabled
              }
            )}
          >
            {children}
          </span>
        )}
      </label>
    );
  }
}

export default Radio;
