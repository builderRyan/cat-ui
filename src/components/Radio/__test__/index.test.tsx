import React from 'react';
import { mount } from 'enzyme';
import Radio from '../index';
import sinon from 'sinon';


describe('Radio', () => {
  // reset dom
  beforeEach(() => {
    document.body.innerHTML = '';
  });
  afterEach(() => {
    document.body.innerHTML = '';
  });

  test('renders correctly', () => {
    const wrapper1 = mount(
      <Radio value="A">radio</Radio>
    );
    const wrapper2 = mount(
      <Radio value="A" checked>radio</Radio>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
  });

  test('Radio is selected when it is clicked', () => {
    const handleChange = sinon.spy();

    const wrapper = mount(
      <Radio value="A" onChange={handleChange}>radio</Radio>
    );

    expect(handleChange.callCount).toBe(0);
    expect(wrapper.state('checked')).toBe(false);

    const input = wrapper.find('input');

    // https://github.com/airbnb/enzyme/issues/216#issuecomment-191114107
    input.simulate('change', { target: { checked: true } });

    expect(handleChange.callCount).toBe(1);
    expect(wrapper.state('checked')).toBe(true);
  });


  test('Disabled radio cannot trigger onChange', () => {
    const handleChange = sinon.spy();

    const wrapper = mount(
      <Radio value="A" onChange={handleChange} disabled>radio</Radio>
    );

    expect(handleChange.callCount).toBe(0);
    expect(wrapper.state('checked')).toBe(false);

    const input = wrapper.find('input');
    input.simulate('change');

    expect(handleChange.callCount).toBe(0);
    expect(wrapper.state('checked')).toBe(false);
  });
});


describe('Radio.Group', () => {
  // reset dom
  beforeEach(() => {
    document.body.innerHTML = '';
  });
  afterEach(() => {
    document.body.innerHTML = '';
  });

  test('renders correctly', () => {
    const wrapper1 = mount(
      <Radio.Group defaultValue='A'>
        <Radio value='A'>A</Radio>
        <Radio value='B'>B</Radio>
        <Radio value='C'>C</Radio>
        <Radio value='D'>D</Radio>
      </Radio.Group>
    );

    expect(wrapper1).toMatchSnapshot();
  });

  test('Able to switch radio(unControlled)', () => {
    const handleChange = sinon.spy();

    const wrapper = mount(
      <Radio.Group defaultValue='A' onChange={handleChange}>
        <Radio value='A'>A</Radio>
        <Radio value='B'>B</Radio>
        <Radio value='C'>C</Radio>
        <Radio value='D'>D</Radio>
      </Radio.Group>
    );

    expect(wrapper.state('value')).toBe('A');
    expect(handleChange.callCount).toBe(0);

    wrapper.find('input').at(1).simulate('change', { target: { checked: true, value: 'B' } });

    expect(wrapper.state('value')).toBe('B');
    expect(handleChange.callCount).toBe(1);
  });

  test('Able to switch radio(unControlled)', () => {
    const handleChange = sinon.spy();

    const wrapper = mount(
      <Radio.Group value='A' onChange={handleChange}>
        <Radio value='A'>A</Radio>
        <Radio value='B'>B</Radio>
        <Radio value='C'>C</Radio>
        <Radio value='D'>D</Radio>
      </Radio.Group>
    );
    expect(handleChange.callCount).toBe(0);
    expect(wrapper.state('value')).toBe('A');

    wrapper.find('input').at(1).simulate('change', { target: { checked: true, value: 'B' } });

    wrapper.setProps({ value: 'B' });
    expect(handleChange.callCount).toBe(1);
    expect(wrapper.state('value')).toBe('B');
  });

  test('When Radio.group has null children, null should not be rendered', () => {
    const wrapper = mount(
      <Radio.Group defaultValue='A'>
        <Radio value='A'>A</Radio>
        <Radio value='B'>B</Radio>
        <Radio value='C'>C</Radio>
        <Radio value='D'>D</Radio>
        {null}
      </Radio.Group>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
