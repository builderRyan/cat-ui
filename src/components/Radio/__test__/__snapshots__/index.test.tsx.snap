// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<Radio
  defaultChecked={false}
  disabled={false}
  prefix="cat"
  value="A"
>
  <label
    className="cat-radio-wrapper"
  >
    <span
      className="cat-radio__inner-wrapper"
    >
      <span
        className="cat-radio__inner"
      />
      <input
        checked={false}
        className="cat-radio"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="A"
      />
    </span>
    <span
      className="cat-radio__label"
    >
      radio
    </span>
  </label>
</Radio>
`;

exports[`Radio renders correctly 2`] = `
<Radio
  checked={true}
  defaultChecked={false}
  disabled={false}
  prefix="cat"
  value="A"
>
  <label
    className="cat-radio-wrapper"
  >
    <span
      className="cat-radio__inner-wrapper"
    >
      <span
        className="cat-radio__inner cat-radio__inner--checked"
      />
      <input
        checked={true}
        className="cat-radio"
        disabled={false}
        onChange={[Function]}
        type="radio"
        value="A"
      />
    </span>
    <span
      className="cat-radio__label"
    >
      radio
    </span>
  </label>
</Radio>
`;

exports[`Radio.Group When Radio.group has null children, null should not be rendered 1`] = `
<RadioGroup
  defaultValue="A"
  disabled={false}
>
  <div
    className="cat-radio-group"
  >
    <Radio
      checked={true}
      defaultChecked={false}
      disabled={false}
      key=".0"
      onChange={[Function]}
      prefix="cat"
      value="A"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner cat-radio__inner--checked"
          />
          <input
            checked={true}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="A"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          A
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".1"
      onChange={[Function]}
      prefix="cat"
      value="B"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="B"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          B
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".2"
      onChange={[Function]}
      prefix="cat"
      value="C"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="C"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          C
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".3"
      onChange={[Function]}
      prefix="cat"
      value="D"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="D"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          D
        </span>
      </label>
    </Radio>
  </div>
</RadioGroup>
`;

exports[`Radio.Group renders correctly 1`] = `
<RadioGroup
  defaultValue="A"
  disabled={false}
>
  <div
    className="cat-radio-group"
  >
    <Radio
      checked={true}
      defaultChecked={false}
      disabled={false}
      key=".0"
      onChange={[Function]}
      prefix="cat"
      value="A"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner cat-radio__inner--checked"
          />
          <input
            checked={true}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="A"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          A
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".1"
      onChange={[Function]}
      prefix="cat"
      value="B"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="B"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          B
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".2"
      onChange={[Function]}
      prefix="cat"
      value="C"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="C"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          C
        </span>
      </label>
    </Radio>
    <Radio
      checked={false}
      defaultChecked={false}
      disabled={false}
      key=".3"
      onChange={[Function]}
      prefix="cat"
      value="D"
    >
      <label
        className="cat-radio-wrapper"
      >
        <span
          className="cat-radio__inner-wrapper"
        >
          <span
            className="cat-radio__inner"
          />
          <input
            checked={false}
            className="cat-radio"
            disabled={false}
            onChange={[Function]}
            type="radio"
            value="D"
          />
        </span>
        <span
          className="cat-radio__label"
        >
          D
        </span>
      </label>
    </Radio>
  </div>
</RadioGroup>
`;
